bin/esi-install-image

```diff
@@ -81,26 +91,32 @@ class ServiceImageManager(object):
             should_remove = True
             if not force:
                 for image in image_set:
-                    imageid = image.id
-                    if enabled_images.has_key(imageid):
+                    if image.id in enabled_images:
                         print >> sys.stderr, "Warning: skipping enabled image."
                         print >> sys.stderr, "Use '--force' to remove it anyway."
                         should_remove = False
 
             if should_remove:
                 for image in image_set:
-                    imageid = image.id
-                    deregister_cmd = "/usr/bin/euca-deregister {0}".format(imageid)
+                    deregister_cmd = "/usr/bin/euca-deregister {0} --region {1} " \
+                                     "-I {2} -S {3} ".format(image.id,
+                                                             self.region,
+                                                             self.get_env_var('AWS_ACCESS_KEY_ID'),
+                                                             self.get_env_var('AWS_SECRET_ACCESS_KEY'))
                     deregister_output = subprocess.Popen(deregister_cmd, stdout=subprocess.PIPE,
                                                             stderr=subprocess.PIPE, shell=True)
```

Comment:
This will result in access keys appearing in `ps` output.  Supply them in the environment instead, either by calling os.setenv beforehand or by passing an `env` dictionary to `Popen`'s initializer.

While you're at it, `shell=True` is incredibly dangerous and it makes this code vulnerable to shell injection.  Please build the command as a list and execute that instead.